src/palimpzest/sets.py

```diff
@@ -60,7 +60,7 @@ def __init__(
         self._limit = limit
         self._cardinality = cardinality
         self._depends_on = [] if depends_on is None else sorted(depends_on)
-        self._nocache = nocache
+        self.cache = cache
```

Comment:
TODO: this should be `self._cache = cache` (the class member should be prefixed with an underscore to keep convention with the other class members).